src/vos/evtree.c

```diff
@@ -2131,6 +2131,21 @@ evt_large_hole_insert(daos_handle_t toh, const struct evt_entry_in *entry)
 	return rc == 0 ? alt_rc : rc;
 }
 
+static inline void
+evt_check_sort_policy(struct evt_context *tcx)
+{
+	/** EVT_FEAT_SORT_DIST can get imbalanced and, under certain conditions, can cause a
+	 *  segfault when the depth is larger than EVT_TRACE_MAX.  Rather than let it get too
+	 *  deep, let's detect cases where it starts getting deep and switch policies.
+	 */
+	if (unlikely(tcx->tc_depth > 5 &&
+		     (tcx->tc_feats & EVT_FEATS_SUPPORTED) == EVT_FEAT_SORT_DIST)) {
```

Comment:
I am afraid that only checking the depth may be not enough to say we are in very imbalance status, the total records count seems another importance factor (but seems we do not maintain that). If it is not imbalance issue caused depth, then what is the overhead for such switch?

Reply:
If the tree depth is 6 (23^6 / 2), we can accommodate ~70 million entries in a balanced tree.  I think it's fairly unlikely to hit this without significant imbalance under any normal use case.   For instance, in the array API, we typically only write one stripe to a given evtree.   The ticket is kind of a worst case scenario and would only result in about 1 million entries in the tree, assuming no aggregation to clean up old entries.

Reply:
I changed it to 6.   In my testing, I don't see other patterns manifesting deep tree growth.  In other words, it's really just this type of I/O patterns that causes the default sort to misbehave.